agent/balrogagent/client.py

```diff
@@ -28,6 +28,14 @@ def get_url(api_root, path):
                 resp.raise_for_status()
                 data["csrf_token"] = resp.headers["X-CSRF-Token"]
 
+        # Workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1471109
+        # In our deployed environments, the Agent doesn't connect to admin over
+        # https, which means it won't send back the session token by default,
+        # which breaks csrf token validation. Changing the cookies to insecure
+        # will let them be sent back, but it's a horrible back.
+        for c in client.cookie_jar:
```

Comment:
Perhaps we could feel better about this if we only did it in the production case, where other design provides protection. If someone runs a local agent against dev admin then secure cookies could work normally; also when/if we fix up ssl for local dev. Does that add real value or is it just marginal ?

Reply:
I think it's a good idea. Isolating the awful/insecure-in-some-contexts hack to where we know it's OK to use makes sense.